### ISSUE TITLE
Pinning/unpinning and adding/removing hashtags from a pinned column crashes advanced web interface

### ISSUE THREAD

**Issue by FiveElementNinja** [mastodon/mastodon#26298]
### Steps to reproduce the problem

1. Make sure you have the advanced interface enabled
2. Search for a hashtag
3. Pin the column
4. You will get the following error message:
```
Due to a bug in our code or a browser compatibility issue, this page could not be displayed correctly.

Try refreshing the page. If that does not help, you may still be able to use Mastodon through a different browser or native app.

Mastodon v4.1.5+nightly-2023-07-31
```
5. Try adding or removing hashtags to an already pinned hashtag column and the error occurs again
6. The error also occurs if you try to unpin a column
7. Sometimes after refreshing the crashed page, the action you were trying to do (pinning, unpinning, adding, removing) will have succeeded. Other times it won't.

### Expected behaviour

Pinning, unpinning, adding, or removing hashtags should not crash the page

### Actual behaviour

The page crashed, throwing an error message

### Mastodon instance

mastodon.social

### Mastodon version

v4.1.5+nightly-2023-07-31

### Browser name and version

Firefox 116.0 (64-bit)

### Operating system

Windows 10 Pro 21H2

### Technical details

Here is the stack trace from the error screen:

```
TypeError: e is not a function


1102/_unsubscribe/<@https://mastodon.social/packs/js/features/hashtag_timeline-6283d2c3f168f339d622.chunk.js:1:87079
_unsubscribe@https://mastodon.social/packs/js/features/hashtag_timeline-6283d2c3f168f339d622.chunk.js:1:87071
componentWillUnmount@https://mastodon.social/packs/js/features/hashtag_timeline-6283d2c3f168f339d622.chunk.js:1:87611
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1181460
pl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180553
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1181636
pl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180553
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1181312
pl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180553
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1181636
pl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180553
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1181312
pl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180553
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1181504
pl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180553
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1181312
pl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180553
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1181636
pl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180553
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180748
pl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180553
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1181504
pl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180553
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1181636
pl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180553
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1181312
pl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180553
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1181312
pl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180553
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1181504
pl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180553
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180748
pl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180553
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180748
pl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180553
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1181504
pl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180553
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1181504
pl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180553
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1181504
pl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180553
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1181504
pl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180553
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1181312
pl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180553
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1181636
pl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180553
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1181312
pl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180553
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1181636
pl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180553
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1181312
pl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180553
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1181504
pl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180553
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1181504
pl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180553
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1181504
pl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180553
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1181312
pl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180553
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1181504
pl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1180553
hl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1181312
ml@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1182124
bl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1182522
ml@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1182274
bl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1183702
ml@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1182274
bl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1182387
ml@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1182274
bl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1183702
ml@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1182274
bl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1182522
ml@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1182274
bl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1182387
ml@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1182274
bl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1182522
ml@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1182274
bl@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1183573
Oc/<@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1197455
Oc@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1198012
cc@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1191768
Ha@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1132468
ac/<@https://mastodon.social/packs/js/common-5ebdb3b06f95668e963a.js:2:1189169



_unsubscribe (webpack:///app/javascript/mastodon/features/hashtag_timeline/index.jsx:124:39)
_unsubscribe (webpack:///app/javascript/mastodon/features/hashtag_timeline/index.jsx:124:21)
componentWillUnmount (webpack:///app/javascript/mastodon/features/hashtag_timeline/index.jsx:160:9)
f.hasOwnProperty (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4640:73)
pl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4578:13)
f.hasOwnProperty (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4640:285)
pl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4578:13)
hl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4633:17)
pl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4578:13)
f.hasOwnProperty (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4640:285)
pl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4578:13)
hl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4633:17)
pl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4578:13)
f.hasOwnProperty (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4640:121)
pl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4578:13)
hl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4633:17)
pl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4578:13)
f.hasOwnProperty (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4640:285)
pl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4578:13)
Cj (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4587:10)
pl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4578:13)
f.hasOwnProperty (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4640:121)
pl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4578:13)
f.hasOwnProperty (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4640:285)
pl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4578:13)
hl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4633:17)
pl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4578:13)
hl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4633:17)
pl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4578:13)
f.hasOwnProperty (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4640:121)
pl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4578:13)
Cj (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4587:10)
pl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4578:13)
Cj (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4587:10)
pl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4578:13)
f.hasOwnProperty (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4640:121)
pl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4578:13)
f.hasOwnProperty (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4640:121)
pl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4578:13)
f.hasOwnProperty (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4640:121)
pl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4578:13)
f.hasOwnProperty (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4640:121)
pl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4578:13)
hl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4633:17)
pl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4578:13)
f.hasOwnProperty (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4640:285)
pl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4578:13)
hl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4633:17)
pl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4578:13)
f.hasOwnProperty (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4640:285)
pl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4578:13)
hl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4633:17)
pl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4578:13)
f.hasOwnProperty (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4640:121)
pl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4578:13)
f.hasOwnProperty (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4640:121)
pl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4578:13)
f.hasOwnProperty (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4640:121)
pl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4578:13)
hl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4633:17)
pl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4578:13)
f.hasOwnProperty (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4640:121)
pl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4578:13)
hl (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4633:17)
removeChild (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4663:139)
vb (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4686:28)
size (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4665:110)
fb (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4755:20)
size (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4665:110)
vb (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4673:19)
size (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4665:110)
fb (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4755:20)
size (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4665:110)
vb (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4686:28)
size (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4665:110)
vb (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4673:19)
size (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4665:110)
vb (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4686:28)
size (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4665:110)
sb (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4750:51)
Oc/< (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:5512:12)
bd (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:5557:32)
lc.onCommitFiberUnmount (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:5139:40)
fa (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:1927:89)
d.hasOwnProperty (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4519:386)
```

**Comment by renchap** [comment]
I am not able to reproduce this on mastodon.social. Can you reproduce it with any Hashtag, like #mastodon?

**Comment by FiveElementNinja** [comment]
> I am not able to reproduce this on mastodon.social. Can you reproduce it with any Hashtag, like #mastodon?

It appears to be fixed now. I can no longer reproduce the error. Perhaps it was related to the recent issue with search?

**Comment by renchap** [comment]
Yes this is probably related, a query to the search server was failing and not properly handled, so it crashed the whole page.

I will keep this open for now as we might want to handle this more gracefuly.

**Comment by FiveElementNinja** [comment]
Much appreciated. 👍

**Comment by ClearlyClaire** [comment]
I am unable to reproduce it, even when purposefully modifying the search controllers to crash.

If the stacktrace is accurate (sometimes it's not because of minimization), this would rather be an issue with streaming (specifically, the `dispatch(connectHashtagStream(…))` call returning `undefined`), but I can't see any code path that would cause it, and I'm also unable to make it fail by stopping the streaming server.

**Comment by FiveElementNinja** [comment]
This issue just started occurring again about 20 minutes ago. However, I _think_ it might be due to this extension I had installed: https://github.com/virtulis/substitoot

Once I removed that extension, the problem went away, though it's hard for me to be certain because even with the extension installed I'm not always able to reproduce the error. I'm guessing that extension was the culprit though. I'm just not clear on why the problem went away the first time and why it's back. Maybe a rate limit thing.

**Comment by ClearlyClaire** [comment]
An extension messing with the DOM is indeed likely to cause issues like this, although I fail to see how that particular error could be triggered. Did you run into it again?

**Comment by FiveElementNinja** [comment]
I stopped using the extension and no longer encountered the error I reported, however, I reinstalled the extension the other day and although I still wasn't encountering the error for a while, I did start getting it today. I don't really know what changed to make the error start occurring, but it first happened after making a post. Now I can cause it to happen using the steps I mentioned in my original post.

Removing the extension makes the problem go away though, so my guess is it's something the extension is doing.
